banyand/measure/query.go

```diff
@@ -141,13 +175,59 @@ func (s *measure) Query(ctx context.Context, mqo pbv1.MeasureQueryOptions) (pbv1
 		return &result, nil
 	}
 
-	result.sidToIndex = make(map[common.SeriesID]int)
-	for i, si := range originalSids {
-		result.sidToIndex[si] = i
-	}
 	return &result, nil
 }
 
+func mustEncodeTagValue(tagType databasev1.TagType, tagValue *modelv1.TagValue, num int) [][]byte {
+	values := make([][]byte, 0)
+	switch tagType {
+	case databasev1.TagType_TAG_TYPE_INT:
+		if tagValue.GetInt() != nil {
+			for i := 0; i < num; i++ {
+				values = append(values, convert.Int64ToBytes(tagValue.GetInt().GetValue()))
+			}
+		}
+	case databasev1.TagType_TAG_TYPE_STRING:
+		if tagValue.GetStr() != nil {
+			for i := 0; i < num; i++ {
+				values = append(values, []byte(tagValue.GetStr().GetValue()))
+			}
+		}
+	case databasev1.TagType_TAG_TYPE_DATA_BINARY:
+		if tagValue.GetBinaryData() != nil {
+			for i := 0; i < num; i++ {
+				values = append(values, bytes.Clone(tagValue.GetBinaryData()))
+			}
+		}
+	case databasev1.TagType_TAG_TYPE_INT_ARRAY:
+		if tagValue.GetIntArray() == nil {
+			return values
+		}
+		for i := 0; i < num; i++ {
+			value := make([]byte, 0)
+			for j := range tagValue.GetIntArray().Value {
+				value = append(value, byte(128))
+				value = append(value, convert.Int64ToBytes(tagValue.GetIntArray().Value[j])...)
+			}
+			values = append(values, value)
+		}
+	case databasev1.TagType_TAG_TYPE_STRING_ARRAY:
+		if tagValue.GetStrArray() == nil {
+			return values
+		}
+		for i := 0; i < num; i++ {
+			var value string
+			for j := range tagValue.GetStrArray().Value {
```

Comment:
The encoding method is incorrect.

According to `block.mustInitFromTagsAndFields`, the encoding process should contain two steps:

1. Convert `tagValue` to `nameValue` using `encodeTagValue` laid in `write.go`
2. `nameValue.marshal` could serialize the value to a byte slice.